Source/DataSources/I3SDataSource.js

```diff
@@ -0,0 +1,4601 @@
+    }
+  } else if (this._data.lodSelection !== undefined) {
+    for (
+      var lodIndex = 0;
+      lodIndex < this._data.lodSelection.length;
+      lodIndex++
+    ) {
+      if (
+        this._data.lodSelection[lodIndex].metricType === "maxScreenThreshold"
+      ) {
+        metersPerPixel = span / this._data.lodSelection[lodIndex].maxError;
+      }
+    }
+  }
+
+  if (metersPerPixel === Infinity) {
+    metersPerPixel = 100000;
+  }
+
+  // calculate the length of 16 pixels in order to trigger the screen space error
+  var geometricError = metersPerPixel * 16;
+
+  // transformations
+  var hpr = new HeadingPitchRoll(0, 0, 0);
+  var orientation = Transforms.headingPitchRollQuaternion(position, hpr);
+
+  if (this._data.obb) {
+    orientation = new Quaternion(
+      this._data.obb.quaternion[0],
+      this._data.obb.quaternion[1],
+      this._data.obb.quaternion[2],
+      this._data.obb.quaternion[3]
+    );
+  }
+
+  this._rotationMatrix = Matrix3.fromQuaternion(orientation);
+  this._inverseRotationMatrix = new Matrix3();
+  Matrix3.inverse(this._rotationMatrix, this._inverseRotationMatrix);
+
+  this._globalTransforms = new Matrix4(
+    this._rotationMatrix[0],
+    this._rotationMatrix[1],
+    this._rotationMatrix[2],
+    0,
+    this._rotationMatrix[3],
+    this._rotationMatrix[4],
+    this._rotationMatrix[5],
+    0,
+    this._rotationMatrix[6],
+    this._rotationMatrix[7],
+    this._rotationMatrix[8],
+    0,
+    position.x,
+    position.y,
+    position.z,
+    1
+  );
+
+  this.inverseGlobalTransform = new Matrix4();
+  Matrix4.inverse(this._globalTransforms, this.inverseGlobalTransform);
+
+  var localTransforms = this._globalTransforms.clone();
+
+  if (this._parent._globalTransforms) {
+    Matrix4.multiply(
+      this._globalTransforms,
+      this._parent.inverseGlobalTransform,
+      localTransforms
+    );
+  }
+
+  // get children definition
+  var childrenDefinition = [];
+  for (var childIndex = 0; childIndex < this._children.length; childIndex++) {
+    childrenDefinition.push(
+      this._children[childIndex]._create3DTileDefinition()
+    );
+  }
+
+  // Create a tile set
+  var inPlaceTileDefinition = {
+    children: childrenDefinition,
+    refine: "REPLACE",
+    boundingVolume: boundingVolume,
+    transform: [
+      localTransforms[0],
+      localTransforms[4],
+      localTransforms[8],
+      localTransforms[12],
+      localTransforms[1],
+      localTransforms[5],
+      localTransforms[9],
+      localTransforms[13],
+      localTransforms[2],
+      localTransforms[6],
+      localTransforms[10],
+      localTransforms[14],
+      localTransforms[3],
+      localTransforms[7],
+      localTransforms[11],
+      localTransforms[15],
+    ],
+    content: {
+      uri: this._completeUriWithoutQuery,
+    },
+    geometricError: geometricError,
+  };
+
+  // Add an entry in the data source content cache
+  _i3sContentCache[this._completeUriWithoutQuery] = {
```

Comment:
The [`Cesium3DTilesetCache`](https://github.com/CesiumGS/cesium/blob/main/Source/Scene/Cesium3DTilesetCache.js) is not fully being utilized here, given that some of the [`Cesium3DTile`](https://github.com/CesiumGS/cesium/blob/main/Source/Scene/Cesium3DTile.js) functions are being overwrriten.  It may be possible to use another cache system here, but as mentioned below, it would ideally store the transcoded [`ModelComponents`](https://github.com/CesiumGS/cesium/blob/main/Source/Scene/ModelComponents.js) and not the raw payload, to avoid retranscoding.